islandora_video_streaming_panopto.module

```diff
@@ -40,20 +57,256 @@ function islandora_video_streaming_panopto_islandora_sp_videoCModel_islandora_de
       $variables['islandora_content'] = $new_content;
+    } 
+  }
+}
+
+/*
+ * Implements hook_libraries_info().
+ * @return array Library needed for upload
+ */
+function islandora_video_streaming_panopto_libraries_info() {
+  $libraries['aws'] = array(
+    'name' => 'aws library', 
+    'vendor url' => 'https://aws.amazon.com/', 
+    'download url' => 'https://docs.aws.amazon.com/aws-sdk-php/v3/download/aws.zip', 
+    'version arguments' => array(
+      'file' => 'README.md',
+      'pattern' => '/Version (\d+)/', 
+    ), 
+    'files' => array(
+      'php' => array('aws-autoloader.php'),
+    ),
+  );
+  return $libraries;
+}
+
+/*
+ * Implements hook_theme()
+ * @return array Everything necessary for iframe.tlp.theme
+ */
+function islandora_video_streaming_panopto_theme() {
+  return array(
+    'panopto_streaming_iframe' => array(
+      'variables' => array(
+        'url' => null,
+        'height' => null,
+        'width' => null,
+        'aspect_ratio' => null,
+        'start_at' => null,
+        'interactivity' => null,
+        'autoplay' => null,
+        'showbrand' => null,
+        'showtitle' => null,
+        'offer_viewer' => null,
+      ),
+      'template' => 'iframe',
+      'path' => drupal_get_path('module','islandora_video_streaming') . DIRECTORY_SEPARATOR. 'theme',
+    )
+  );
+} 
+
+/*
+ * Implements hook_islandora_object_alter().
+ * @param AbstractObject $object The object being altered
+ * @param array $context The context or action of which the object will be altered
+ */
+function islandora_video_streaming_panopto_islandora_object_alter(AbstractObject $object, &$context) {
+  islandora_video_streaming_panopto_delete_from_panopto($object, $context);
+} 
+
+/*
+ * Implements hook_islandora_datastream_alter().
+ * @param AbstractObject $object The object being altered
+ * @param AbstractDatastream $datastream The datastream being altered
+ * @param array $context The context or action of which the object will be altered
+ */
+function islandora_video_streaming_panopto_islandora_datastream_alter(AbstractObject $object, AbstractDatastream $datastream, &$context) {
+  if ($datastream->id == 'PANOPTO_INFO') {
+    islandora_video_streaming_panopto_delete_from_panopto($object, $context);
+  }
+}
+
+/*
+ * Deletes video from Panopto server.
+ * @param AbstractObject $object The object being altered
+ * @param array $context The context or action of which the object will be altered
+ */
+function islandora_video_streaming_panopto_delete_from_panopto(AbstractObject $object, &$context) {
+  if (isset($object['PANOPTO_INFO']) && ($context['action'] == 'purge' || $context['action'] == 'delete' )) {
+   //Panopto/api/v1/sessions/{id} for deleting videos on Panopto
+    $decoded = json_decode($object['PANOPTO_INFO']->content);
+    if (isset($decoded)) {
+      $id = $decoded->upload_id;
+      if (isset($id)) {
+        $oauthendpoint = variable_get('islandora_video_streaming_panopto_host');
```

Comment:
As before, this is more accurately, a `$host` variable which is then used to construct the `$oauthendpoint` and the `$deleteendpoint` (or `$objectendpoint`).

Reply:
Fixed.